src/meta/api/src/schema_api_impl.rs

```diff
@@ -2525,6 +2533,162 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
         Ok(CountTablesReply { count })
     }
 
+    async fn list_table_lock_revs(
+        &self,
+        req: ListTableLockRevReq,
+    ) -> Result<ListTableLockRevReply, KVAppError> {
+        let prefix = format!("{}/{}", TableLockKey::PREFIX, req.table_id);
+        let reply = self.prefix_list_kv(&prefix).await?;
+
+        let mut revisions = vec![];
+        for (k, _) in reply.into_iter() {
+            let lock_key = TableLockKey::from_str_key(&k).map_err(|e| {
+                let inv = InvalidReply::new("list_table_lock_revs", &e);
+                let meta_net_err = MetaNetworkError::InvalidReply(inv);
+                MetaError::NetworkError(meta_net_err)
+            })?;
+
+            revisions.push(lock_key.revision);
+        }
+        // sort ascending.
+        revisions.sort();
+        Ok(ListTableLockRevReply { revisions, prefix })
+    }
+
+    async fn upsert_table_lock_rev(
+        &self,
+        req: UpsertTableLockRevReq,
+    ) -> Result<UpsertTableLockRevReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let mut retry = 0;
+        let table_id = req.table_id;
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
+
+            let tbid = TableId { table_id };
+            let (tb_meta_seq, tb_meta): (_, Option<TableMeta>) = get_pb_value(self, &tbid).await?;
+            if tb_meta_seq == 0 {
+                return Err(KVAppError::AppError(AppError::UnknownTableId(
+                    UnknownTableId::new(table_id, "upsert_table_lock_rev"),
+                )));
+            }
+
+            debug!(
+                ident = display(&tbid),
+                table_meta = debug(&tb_meta),
+                "upsert_table_lock_rev"
+            );
+
+            let revision = if let Some(rev) = req.revision {
+                rev
+            } else {
+                // generate a new revision.
+                fetch_id(self, IdGenerator::table_lock_id()).await?
+            };
+
+            let lock_key = TableLockKey { table_id, revision };
+
+            let (lock_key_seq, _): (_, Option<TableLock>) = get_pb_value(self, &lock_key).await?;
+
+            if lock_key_seq > 0 && req.revision.is_none() {
+                // lock key has been created by other process.
+                continue;
+            }
+
+            let lock = TableLock {};
+
+            let condition = vec![txn_cond_seq(&lock_key, Eq, lock_key_seq)];
```

Comment:
if update lock need to assure that table is exist?

Reply:
> if update lock need to assure that table is exist?

Do we need to check the table mete sequence (version)  here? @dantengsky